crates/jstzd/src/container.rs

```diff
@@ -0,0 +1,114 @@
+use async_dropper::{AsyncDrop, AsyncDropper};
+use async_trait::async_trait;
+use bollard::{container::StartContainerOptions, Docker};
+use std::sync::Arc;
+
+use crate::network::Network;
+
+pub type ContainerId = String;
+#[derive(Default)]
+struct ContainerInner {
+    pub id: ContainerId,
+    pub name: String,
+    client: Option<Arc<Docker>>,
+    // TODO: network: Option<Network>,
+    _private: (),
+}
+
+pub struct Container(AsyncDropper<ContainerInner>);
+
+impl ContainerInner {
+    /// Creates a new container with running `id`
+    pub fn new(
+        client: Arc<Docker>,
+        id: ContainerId,
+        name: &str,
+        _network: Option<Network>,
+    ) -> Self {
+        Self {
+            id,
+            name: name.to_string(),
+            client: Some(client),
+            // network,
+            _private: (),
+        }
+    }
+
+    /// Starts the container's entrypoint
+    pub async fn start(&self) -> anyhow::Result<()> {
+        let client = self.client()?;
+        if let Some(state) = client.inspect_container(&self.name, None).await?.state {
+            if let Some(true) = state.running {
+                println!("Container is already running");
+                return Ok(());
+            }
+        }
+        client
+            .start_container(&self.name, None::<StartContainerOptions<String>>)
```

Comment:
The container name should be unique similar to the id, since the `name` is required to create a container (in RunnableImage::create_container), if we just use the `name` we can remove the extra field `id` from the container as well? wdyt

Reply:
See discussion above about `id`. Still recommend using `id` over `name`